client/core/core.go

```diff
@@ -1933,18 +1956,37 @@ func (c *Core) initialize() {
 		wg.Add(1)
 		go func(acct *db.AccountInfo) {
 			defer wg.Done()
+			if acct.Disabled {
+				return
+			}
+			host := addrHost(acct.Host)
 			dc, err := c.connectDEX(acct)
 			if err != nil {
 				log.Errorf("error connecting to DEX %s: %v", acct.Host, err)
 				return
 			}
 			log.Debugf("connectDEX for %s completed, checking account...", acct.Host)
+			if c.conns[host] != nil && c.conns[host].acct != nil {
```

Comment:
When does the condition `c.conns[host].acct != nil` happen?  It's not clear to me how a `dexConnection` with a nil `acct` field would get into the `c.conns` map.

Reply:
This occurred for me when I was testing and the disable account action did not happen when it was supposed to. I'm not sure it will ever happen in the real world but it did happen to me & the only way to fix was to manually edit data so I coded the fix.

Reply:
Sounds like your tests needed more setup. A `dexConnection` without a `dexAccount` is not possible in production, and if it happened, we'd want to `panic` anyways, so there's really no need to check here. Generally, it's not good practice to change the code to fit your test.

Reply:
For clarification the situation occurred from manual testing, I do not change code to fit tests.